core/Files.cc

```diff
@@ -174,8 +174,18 @@ bool isPackageRBIPath(string_view path) {
     return absl::EndsWith(path, ".package.rbi");
 }
 
+bool isPackagePath(string_view path) {
+    auto pos = path.rfind("/");
+    if (pos != string_view::npos) {
+        path = path.substr(pos + 1);
+    }
+
+    return path == "__package.rb";
+}
```

Comment:
We used to only set `sourceType = Type::Package` in `packager.cc`, which would have only been run if the `--stripe-packages` flag had been passed at the CLI.

Do you think it's possible to thread that through here?

(It would be an ever so slight backwards-incompatible change in that if your project happens to have a `__package.rb` file already.)

Reply:
I think it's the case already that any path that cares about `isPackage` would be gated by `--stripe-packages` already, right? This change makes us even less likely to accidentally treat a package file specially, as they now all have the sourceType `File::Type::Normal`.

Reply:
Ah good point. Thanks!

Reply:
Separately, I don't think it would be tough to thread that through to this point, so maybe to guard against a possible foot-gun I'll add that.

Reply:
up to you—change looks good either way

Reply:
I take it back -- I thought we had a boolean in `GlobalState` for signaling when `--stripe-packages` is enabled and we don't. Unfortunately we'd probably need to add a flag like that to `GlobalState` to make this easy.

Reply:
Looking through the changes to LSP, I think this might handle `__package.rb` files differently even without `--stripe-packages`.  I'm going to see if I can convince myself otherwise before merging this.

Reply:
I've guarded the uses of `isPackage()` in LSP with `this->config.opts.stripePackages`, which I think addresses any issues that would have arisen from this change.